### PR TITLE
Bug safer regex

### PR DESCRIPTION
**Pull request by seabbs**
Safer regex as suggested by @sbfnk here: https://github.com/epiforecasts/epinowcast/pull/85#discussion_r913227210

### CODE REVIEW

**Comment by seabbs** [comment]
Hmm a problem for tomorrow. Looks like it isn't getting the last bracket. 

```r
`actual`:   "vector[n] x[nl, pq;" 
`expected`: "vector[n] x[nl, pq];"
```

File: R/utils.R [hunk 1]
```diff
@@ -43,7 +43,7 @@ convert_cmdstan_to_rstan <- function(functions) {
   )
   #   case 2: array[n] real x -> real x[n]
   functions <- gsub(
-    "array\\[(.*)\\] (.*) ([a-z_]+)", "\\2 \\3[\\1]", functions
+    "array\\[([^]]*)\\]\\s+(.*)\\s+([a-z_]+)", "\\2 \\3[\\1", functions
```
Comment (sbfnk) [inline R/utils.R:46]: missing a closing `]` before the last quotation mark

Also I think you need to replace `(.*)` with `([a-z_]*)` otherswise you get the same greedy matching issues as before.

Comment (seabbs) [reply R/utils.R:46]: That won't account properly for vector variables will it?

Comment (sbfnk) [reply R/utils.R]: I must be missing something. Can you give an example where it wouldn't?

Comment (seabbs) [reply R/utils.R]: I've added another line of regex to crudely deal with this. I think perhaps should fix it as bugs occur now? Also added a few harder tests for things not currently in functions but that could be. See https://github.com/epiforecasts/epinowcast/pull/93/commits/5953dd18d7389499a84a218defe483212d4e04ba

**Pull request merged**